### ISSUE TITLE
display.flip()  Gives Window surface is invalid error

### ISSUE THREAD

**Issue by mossaybo** [pygame/pygame#1698]
**WARNING** : I'm totally new to pygame and relatively new to python , I did my research on the problem but I'm stuck.
with that said, the following code will generate a surface, while waiting for QUIT event keeps refreshing the screen using `pygame.display.flip()`
```
import sys
import pygame
def run_game():
    pygame.init()
    screen = pygame.display.set_mode((1200, 800))
    pygame.display.set_caption("Learning Pygame")
    
    #the game loop/ to update surfaces AND listen to events
    while True:
        #when ever an event (eg key press) is caused , the loop will trigger
        #all you need to do is to filter type of event using if
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                sys.exit()
        
        #then update screen
        pygame.display.flip()

run_game()
```
the following error will result 
```
pygame 2.0.0.dev6 (SDL 2.0.10, python 3.7.6)
Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
  File "make_window.py", line 19, in <module>
    run_game()
  File "make_window.py", line 17, in run_game
    pygame.display.flip()
pygame.error: Window surface is invalid, please call SDL_GetWindowSurface() to get a new surface
```

I confirm that **the code works fine in** `pygame 1.9.6` but due to high CPU usage I had to update .

**More informations** 
> Distro: MX-18.1_x64
> Wm: i3-gaps

**Comment by MightyJosip** [comment]
Hi, thanks for your report. Can you tell us do you do anything before the game crashes, or it crash the moment you run it. Also what OS do you have

**Comment by mossaybo** [comment]
> Hi, thanks for your report. Can you tell us do you do anything before the game crashes, or it crash the moment you run it.

Hi, thanks for your reply .
I do nothing , it crushes directly the moment I run it .

**Comment by MightyJosip** [comment]
There were a few changes recently to the flip function that could resolve this problem. We will soon release pygame 2.0.0.dev8. I will mention you here once we do it and tell you how to install it. One last question from me. Do you get the same error with the following code:

```import sys
import pygame
import time


def run_game():
    pygame.init()
    screen = pygame.display.set_mode((1200, 800))
    pygame.display.set_caption("Learning Pygame")

    time.sleep(1)
    # the game loop/ to update surfaces AND listen to events
    while True:
        # when ever an event (eg key press) is caused , the loop will trigger
        # all you need to do is to filter type of event using if
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                sys.exit()

        # then update screen
        pygame.display.flip()


run_game()```

**Comment by mossaybo** [comment]
Ok thank you .
As for the first code : the test **failed** (same traceback)
the second code **works fine** (high cpu usage but maybe because of the loop)

**TL;DR** 
for people having same issue (Window surface is invalid, please call SDL_GetWindowSurface)
1- replace `display.flip()`  with `pygame.display.update()`
OR
2- downgrade to `pygame 1.9.6` and use `pygame.mixer.quit()` to fix high cpu usage problem

**Comment by MightyJosip** [comment]
To reduce cpu usage use clock class

**Comment by mossaybo** [comment]
> To reduce cpu usage use clock class

I'm sorry I don't want to take much of your time , but if it's straightforward can you make the proper change for that in my code .

**Comment by MightyJosip** [comment]
```
import sys
import pygame

clock = pygame.time.Clock()
FPS = 60
def run_game():
    pygame.init()
    screen = pygame.display.set_mode((1200, 800))
    pygame.display.set_caption("Learning Pygame")

    # the game loop/ to update surfaces AND listen to events
    while True:
        # when ever an event (eg key press) is caused , the loop will trigger
        # all you need to do is to filter type of event using if
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                sys.exit()

        # then update screen
        pygame.display.update()
        clock.tick(60)

run_game()
```
If you would like you can also join pygame discord server for help (https://discord.gg/r8yreB6).

**Comment by mossaybo** [comment]
Really appreciate it , THANK YOU !
I'll make sure to register in discord and join pygame server if I have any future problems .
Salam

**Comment by MightyJosip** [comment]
@mossaybo Can you install the newest pygame version with `pip install pygame==2.0.0.dev8` and then run the code with `pygame.display.flip()` statement. If yes can you say do you still get the same error.

**Comment by mossaybo** [comment]
> @mossaybo Can you install the newest pygame version with `pip install pygame==2.0.0.dev8` and then run the code with `pygame.display.flip()` statement. If yes can you say do you still get the same error.

sure, (sorry for the delay)
so I tried the my first code and I confirm **it's working fine** 
problem **SOLVED**
thank you.

```
import sys
import pygame
def run_game():
    pygame.init()
    screen = pygame.display.set_mode((1200, 800))
    pygame.display.set_caption("Learning Pygame")
    
    #the game loop/ to update surfaces AND listen to events
    while True:
        #when ever an event (eg key press) is caused , the loop will trigger
        #all you need to do is to filter type of event using if
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                sys.exit()
        
        #then update screen
        pygame.display.flip()

run_game()
```
